botorch/acquisition/multi_objective/joint_entropy_search.py

```diff
@@ -0,0 +1,746 @@
+                containing the sampled Pareto optimal sets of outputs.
+            hypercell_bounds:  A `num_pareto_samples x 2 x J x M`-dim Tensor
+                containing the hyper-rectangle bounds for integration, where `J` is
+                the number of hyper-rectangles. In the unconstrained case, this gives
+                the partition of the dominated space. In the constrained case, this
+                gives the partition of the feasible dominated space union the
+                infeasible space.
+            X_pending: A `m x d`-dim Tensor of `m` design points that have been
+                submitted for function evaluation, but have not yet been evaluated.
+            estimation_type: A string to determine which entropy estimate is
+                computed: "0", "LB", "LB2", or "MC".
+            num_samples: The number of Monte Carlo samples for the Monte Carlo
+                estimate.
+            condition_on_samples: If true, we condition on the Pareto optimal
+                points.
+        """
+        super().__init__(model=model)
+        # Batch GP models (e.g. fantasized models) are not currently supported
+        if isinstance(model, ModelListGP):
+            train_X = model.models[0].train_inputs[0]
+        else:
+            train_X = model.train_inputs[0]
+        if (model.num_outputs > 1 and train_X.ndim > 3) or (
+            model.num_outputs == 1 and train_X.ndim > 2
+        ):
+            raise NotImplementedError(
+                "Batch GP models (e.g. fantasized models) are not supported."
+            )
+
+        self.initial_model = model
+        if (pareto_sets is not None and pareto_sets.ndim != 3) or (
+            pareto_fronts is not None and pareto_fronts.ndim != 3
+        ):
+            raise UnsupportedError(
+                "The Pareto set and front should have a shape of "
+                "`num_pareto_samples x num_pareto_points x input_dim` and "
+                "`num_pareto_samples x num_pareto_points x num_objectives`, "
+                "respectively"
+            )
+        else:
+            self.pareto_sets = pareto_sets
+            self.pareto_fronts = pareto_fronts
+
+        if hypercell_bounds.ndim != 4:
+            raise UnsupportedError(
+                "The hypercell_bounds should have a shape of "
+                "`num_pareto_samples x 2 x num_boxes x num_objectives`."
+            )
+        else:
+            self.hypercell_bounds = hypercell_bounds
+            self.num_pareto_samples = hypercell_bounds.shape[0]
+
+        # Condition the model on the sampled pareto optimal points.
+        # TODO: Apparently, we need to make a call to the posterior otherwise
+        #  we run into a gpytorch runtime error:
+        #  "Fantasy observations can only be added after making predictions with a
+        #  model so that all test independent caches exist."
+
+        if condition_on_samples:
```

Comment:
This is specific for ` qLowerBoundMultiObjectiveJointEntropySearch`? If so, let's move it into `qLowerBoundMultiObjectiveJointEntropySearch`